include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -150,6 +153,54 @@ __pattern_transform_scan_base(_ExecutionPolicy&& __exec, _Iterator1 __first, _It
         // global scan
         unseq_backend::__global_scan_functor<_Inclusive, _BinaryOperation, _InitType>{__binary_op, __init})
         .wait();
+}
+
+template <typename _ExecutionPolicy, typename _Iterator1, typename _Iterator2, typename _UnaryOperation,
+          typename _InitType, typename _BinaryOperation, typename _Inclusive>
+oneapi::dpl::__internal::__enable_if_hetero_execution_policy<_ExecutionPolicy, _Iterator2>
+__pattern_transform_scan_base(_ExecutionPolicy&& __exec, _Iterator1 __first, _Iterator1 __last, _Iterator2 __result,
+                              _UnaryOperation __unary_op, _InitType __init, _BinaryOperation __binary_op, _Inclusive)
+{
+    if (__first == __last)
+        return __result;
+
+    const auto __n = __last - __first;
+
+    const bool bInplaceExclusiveScan = __n > 1 && !_Inclusive{} && __first == __result;
+    if (!bInplaceExclusiveScan)
+    {
+        __pattern_transform_scan_base_impl(__exec, __first, __last, __result, __unary_op, __init, __binary_op,
+                                           _Inclusive{});
+    }
+    else
+    {
+        // This is a temporary workaround for in-place exclusive scan so far the SYCL backend scan pattern has not fixed.
+        assert(__n > 1);
+        assert(!_Inclusive{});
+        assert(__result == __first);
+
+        using _Type = typename _InitType::__value_type;
+
+        auto __exec_tmp =
```

Comment:
lets rename to `__policy ` for a example. (at least suffix tmp confuses here I think)

Reply:
Fixed